docs/guides/fixtures/README.md

```diff
@@ -0,0 +1,13 @@
+# Introduction to Fixtures
+
+In a Bocadillo application, views typically need to access _resources_ or _services_ to fulfill a client's request. For example, they may need to query a database, or interact with a disk-based cache.
+
+Yet, views shouldn't have to care about _how_ those resources are assembled, provided, nor cleaned up.
+
+To address this issue, **fixtures** allow resources to be _injected_ into the view. They are inspired by [pytest fixtures](https://docs.pytest.org/en/latest/fixture.html) and are an example of [Dependency Injection](https://en.wikipedia.org/wiki/Dependency_injection).
+
+Fixtures aim at being:
+
+- **Explicit**: once a fixture is defined, it can be consumed by an HTTP or WebSocket view simply by declaring it as a function parameter.
+- **Modular**: a fixture can itself use other fixtures.
+- **Flexible**: fixtures are reusable within the scope of an app or a session, and support a variety of syntaxes (asynchronous vs. synchronous, function vs. generator) to make provisioning resources fun again.
```

Comment:
Thoughts @waghanza? :-)

Reply:
I think it could be cleaner (in my point of view) to use them from decorators

Reply:
Which decorators do you refer to? I’m not sure I understood what you meant.

Reply:
The purpose is not to have it in function parameters, it could add some noise.
It could be accurate to use them like this
~~~python
class MyView
  @app.fixtures(fixture_one,fixture_two)
  def bar:
~~~

Reply:
I see! But then:

- You would have to explicitly import them from somewhere, right? (whereas the proposed system does not require any import, so long as the fixture has been discovered)
- It’s not really clear (read: explicit) that the function takes those as inputs, except if we also add them as parameters to the function, which becomes quite redundant.
- Note how the proposed design does not tie fixtures to the app at all. This is intended, as in the past cramming everything into the App class has shown to induce high coupling where it wasn’t needed. Take the old app.template syntax for example: it was moved to a separate Templates helper so that templates could be used outside an app, if needed.

Reply:
There is a couple of workaround here 
+ To avoid explicit import on each fixtures, the fixture decorator could have a **dynamic** import feature
+ Agreed
+ Sure, this add coupling. The fixture helper could be loaded on the end-user implementation

Reply:
The issue with dynamic imports (was I understand them here) would be that we’d be tied to the file system/python path, and then all sorts of dark magic could happen and we’d have to deal with that.

I really appreciate your feedback on this, but I think the proposed API is more suited to the problem at hand. Plus, it’s been in use in Pytest for years and the community (Pytest users at least) has become familiar with it, which is something we can/should leverage.

Reply:
You're right, it's better to avoid this after thinking about

I'll check the implementation before 😊